src/variables/unit-commitment.jl

```diff
@@ -1,31 +1,31 @@
 export add_unit_commitment_variables!
 
 """
-    add_unit_commitment_variables!(model, dataframes, sets)
+    add_unit_commitment_variables!(model, ...)
 
-Adds unit commitment variables to the optimization `model` based on the `:units_on` dataframe.
+Adds unit commitment variables to the optimization `model` based on the `:units_on` indices.
 Additionally, variables are constrained to be integers based on the `sets` structure.
 
 """
-function add_unit_commitment_variables!(model, dataframes, sets)
-    df_units_on = dataframes[:units_on]
+function add_unit_commitment_variables!(model, sets, variables)
+    units_on_indices = variables[:units_on].indices
 
-    model[:units_on] =
-        df_units_on.units_on = [
-            @variable(
-                model,
-                lower_bound = 0.0,
-                base_name = "units_on[$(row.asset),$(row.year),$(row.rep_period),$(row.timesteps_block)]"
-            ) for row in eachrow(df_units_on)
-        ]
+    variables[:units_on].container = [
+        @variable(
+            model,
+            lower_bound = 0.0,
+            base_name = "units_on[$(row.asset),$(row.year),$(row.rep_period),$(row.timesteps_block)]"
+        ) for row in eachrow(units_on_indices)
+    ]
 
     ### Integer Unit Commitment Variables
-    for row in eachrow(df_units_on)
-        if !(row.asset in sets.Auc_integer[row.year])
-            continue
-        end
-
-        JuMP.set_integer(model[:units_on][row.index])
+    integer_units_on_indices = filter(
```

Comment:
The following code is more efficient, but we need to ensure that the indices dataframe has the correct columns, which doesn't happen when there is no assets using the unit commitment. If we ensure that the indices table always has the required columns, no matter if the rows are zero, then we can use this code instead of the current changes (using filter).

```suggestion
    integer_units_on_indices = DataFrames.subset(
        units_on_indices,
        [:asset, :year] => DataFrames.ByRow((a, y) -> a in sets.Auc_integer[y]),
    )
```